flit/install.py

```diff
@@ -329,33 +330,17 @@ def install_directly(self):
 
     def install_with_pip(self):
         self.install_reqs_my_python_if_needed()
-
-        with tempfile.TemporaryDirectory() as td:
-            temp_whl = osp.join(td, 'temp.whl')
-            with open(temp_whl, 'w+b') as fp:
-                wb = WheelBuilder(
-                    str(self.directory),
-                    self.module,
-                    metadata=common.make_metadata(self.module, self.ini_info),
-                    entrypoints=self.ini_info.entrypoints,
-                    target_fp=fp,
-                )
-                wb.build()
-
-            renamed_whl = osp.join(td, wb.wheel_filename)
-            os.rename(temp_whl, renamed_whl)
-            extras = self._extras_to_install()
-            extras.discard('.none')
-            whl_with_extras = '{}[{}]'.format(renamed_whl, ','.join(extras)) \
-                if extras else renamed_whl
-
-            cmd = [self.python, '-m', 'pip', 'install', whl_with_extras]
-            if self.user:
-                cmd.append('--user')
-            if self.deps == 'none':
-                cmd.append('--no-deps')
-            shell = (os.name == 'nt')
-            check_call(cmd, shell=shell)
+        extras = self._extras_to_install()
+        extras.discard('.none')
+        req_with_extras = '{}[{}]'.format(self.directory, ','.join(extras)) \
+            if extras else str(self.directory)
+        cmd = [self.python, '-m', 'pip', 'install', req_with_extras]
```

Comment:
Let's have a comment to explain about using pip to do the installation, and that it will install `flit_core` in a temporary environment and invoke it there to make a wheel.

Reply:
I added some explanations in the docstring.